hoomd/md/EvaluatorPairExpandedLJ.h

```diff
@@ -0,0 +1,227 @@
+//! Class for evaluating the Expanded LJ pair potential
+/*! <b>General Overview</b>
+
+    See EvaluatorPairExpandedLJ
+
+    <b>ExpandedLJ specifics</b>
+
+    EvaluatorPairExpandedLJ evaluates the function:
+    \f{eqnarray*}
+    V_{\mathrm{SLJ}}(r)  = & 4 \varepsilon \left[ \left( \frac{\sigma}{r - \Delta} \right)^{12} -
+                           \left( \frac{\sigma}{r - \Delta} \right)^{6} \right] & r <
+   (r_{\mathrm{cut}}) \\
+                         = & 0 & r \ge (r_{\mathrm{cut}}) \\
+    \f}
+
+    The ExpandedLJ potential does not need charge or diameter. Three parameters are specified and
+   stored in a Scalar3. \a lj1 is placed in \a params.x, \a lj2 is in \a params.y, and \a delta
+   is in params.z.
+
+
+    These are related to the standard lj parameters sigma and epsilon by:
+    - \a lj1 = 4.0 * epsilon * pow(sigma,12.0);
+    - \a lj2 = 4.0 * epsilon * pow(sigma,6.0);
+    - \a delta = delta
+
+    Due to the way that ExpandedLJ modifies the cutoff condition, it will not function properly with
+   the xplor shifting mode.
+*/
+class EvaluatorPairExpandedLJ
+    {
+    public:
+    //! Define the parameter type used by this pair potential evaluator
+    struct param_type
+        {
+        Scalar sigma_3;
+        Scalar sigma_6;
+        Scalar epsilon_x_4;
+        Scalar delta;
+
+        DEVICE void load_shared(char*& ptr, unsigned int& available_bytes) { }
+
+        HOSTDEVICE void allocate_shared(char*& ptr, unsigned int& available_bytes) const { }
+
+#ifdef ENABLE_HIP
+        //! Set CUDA memory hints
+        void set_memory_hint() const
+            {
+            // default implementation does nothing
+            }
+#endif
+
+#ifndef __HIPCC__
+        param_type() : sigma_6(0), epsilon_x_4(0), delta(0) { }
+
+        param_type(pybind11::dict v, bool managed = false)
+            {
+            auto sigma(v["sigma"].cast<Scalar>());
+            auto epsilon(v["epsilon"].cast<Scalar>());
+            delta = v["delta"].cast<Scalar>();
+
+            sigma_3 = sigma * sigma * sigma;
+            sigma_6 = sigma_3 * sigma_3;
+            epsilon_x_4 = Scalar(4.0) * epsilon;
+
+            // previous parameters
+            // lj1 = 4.0 * epsilon * pow(sigma, 12.0);
+            // - > lj1 = epsilon_x_4 * sigma_6 * sigma_6
+
+            // lj2 = 4.0 * epsilon * pow(sigma, 6.0);
+            // - > lj2 = epsilon_x_4 * sigma_6
+            }
+
+        // this constructor facilitates unit testing
+        param_type(Scalar sigma, Scalar epsilon, Scalar delta, bool managed = false)
+            {
+            sigma_3 = sigma * sigma * sigma;
+            sigma_6 = sigma_3 * sigma_3;
+            epsilon_x_4 = Scalar(4.0) * epsilon;
+            }
+
+        pybind11::dict asDict()
+            {
+            pybind11::dict v;
+            v["sigma"] = pow(sigma_6, 1. / 6.);
+            v["epsilon"] = epsilon_x_4 / 4.0;
+            v["delta"] = delta;
+            return v;
+            }
+#endif
+        };
```

Comment:
```suggestion
        } __attribute__((aligned(16)));
```
Improve memory access performance on the GPU.